pandas/compat/numpy/function.py

```diff
@@ -245,3 +253,48 @@ def validate_transpose_for_generic(inst, kwargs):
             msg += " for {klass} instances".format(klass=klass)
 
         raise ValueError(msg)
+
+
+def validate_window_func(name, args, kwargs):
+    numpy_args = ('axis', 'dtype', 'out')
+    msg = ("numpy operations are not "
+           "valid with window objects. "
+           "Call the object's '{func}' "
+           "method directly instead".format(func=name))
```

Comment:
maybe construct these error messages a bit more, e.g.

If I had called: ``np.mean(df.rolling(...))`` expecting it to work:

``.rolling(...).mean()`` instead (so a user could almost copy / paste it. 

IOW, much like the depreation warnings that we give for Resampler.